skipchain/skipchain.go

```diff
@@ -367,6 +369,11 @@ func (s *Service) StoreSkipBlockInternal(psbd *StoreSkipBlock) (*StoreSkipBlockR
 				"Couldn't get forward signature on block: " + err.Error())
 		}
 
+		// Set the server's set of valid peers for this skipchain with the new
+		// roster.
+		SetSkipChainValidPeers(s.ServiceProcessor.Context, scID,
```

Comment:
Here you do `SetSkipChainValidPeers` twice: once here, and once in `catchUp`. I propose you keep the one here, and remove the one from `catchUp`.

Reply:
I believe I tried something like that at first, but it seemed in the tests that only the first node (the one receiving the request) was updating the list of valid peers, the other ones only doing it much later. My understanding was that they were doing it during the catch-up, hence the idea to have `catchUp()` set the valid peers as well, which (still per my understanding) also covers the `startAllChains()` case.
Is my understanding incorrect?

Reply:
I think I need holidays, I can't think straight - I mixed `catchUp` with `updateTrieCallback`. 

Did you try to put it `updateTrieCallback` for new blocks and `startAllChains` for startup?

Reply:
Done